sound/soc/sof/sof-client-ipc-msg-injector.c

```diff
@@ -136,8 +137,9 @@ static int sof_msg_inject_probe(struct auxiliary_device *auxdev,
 	if (!priv)
 		return -ENOMEM;
 
-	priv->tx_buffer = devm_kmalloc(dev, SOF_IPC_MSG_MAX_SIZE, GFP_KERNEL);
-	priv->rx_buffer = devm_kzalloc(dev, SOF_IPC_MSG_MAX_SIZE, GFP_KERNEL);
+	priv->max_msg_size = sof_client_get_ipc_max_payload_size(cdev);
```

Comment:
can there be an error here?

Reply:
No. If the max_payload_size would be invalid then the SOF stack would have failed already so we would not have this (or other) clients loaded.